src/docs.js

```diff
@@ -1,34 +1,53 @@
 let tar = require('tar-stream');
 let _ = require('lodash');
 let assert = require('assert');
-let fs = require('fs');
+let fs = require('mz/fs');
 let path = require('path');
+let recursiveReadSync = require('recursive-readdir-sync');
 
 async function documenter(options) {
 
   options = _.defaults({}, options, {
     schemas: [],
+    docsFolder: null,
   });
   assert(options.schemas, 'options.schemas must be given');
   assert(options.schemas instanceof Array, 'options.schemas must be an array');
 
   let schemas = options.schemas;
   let tarball = tar.pack();
 
-  //add schemas to tarball
+  // add schemas to tarball
   schemas.forEach(schema => {
     let data = JSON.stringify(schema, null, 2);
     tarball.entry({name: 'schema/' + schema.id}, data);
   });
 
+  // add docs to tarball
+  let docs = options.docsFolder;
+  let files = recursiveReadSync(options.docsFolder);
+
+  await Promise.all(files.map(file => {
+    //remove the path
+    let relativePath = file.replace(`${options.docsFolder}/`, '');
+    let fileReadPromise = new Promise(function(fulfill, reject) {
+      fs.readFile(file, 'utf8', (err, data) => {
```

Comment:
It's super easy to make mistakes when using new Promise... because you can
easily get in to cases where an error is dropped...
Using async/await this is much harder... so even if you do make a typo, at
least that error will show up... less risk of silent errors...

For the thing you're doing don't return promises to async iterator
function, and await the result of an async operation that returns a
promise...

Note: feel free to copy the snippet into etherpad and ping me...and can
quickly show how to do it...
PS. It's not that what you did won't work or that it contains any bugs :)
it's just that using different patterns, then one day when you do make a
bug... that bug would be a lot easier to find...
Den 8. jul. 2016 12.36 PM skrev "kristelteng" <notifications@github.com>:

> In src/docs.js
> <https://github.com/taskcluster/taskcluster-lib-docs/pull/6#discussion_r70129219>
> :
>
> >    schemas.forEach(schema => {
> >      let data = JSON.stringify(schema, null, 2);
> >      tarball.entry({name: 'schema/' + schema.id}, data);
> >    });
> >
> > +  // add docs to tarball
> > +  let docs = options.docsFolder;
> > +  let files = recursiveReadSync(options.docsFolder);
> > +
> > +  await Promise.all(files.map(file => {
> > +    //remove the path
> > +    let relativePath = file.replace(`${options.docsFolder}/`, '');
> > +    let fileReadPromise = new Promise(function(fulfill, reject) {
> > +      fs.readFile(file, 'utf8', (err, data) => {
>
> since I'm completely new to using promises, can you please explain to me
> how it creates lots of bugs?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/taskcluster/taskcluster-lib-docs/pull/6/files/4e229516cbac8a006bd621bab2537e1b154a3b8e#r70129219>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAJI5N65fRRT793wiW3GyjIjpzUnakQEks5qTqa2gaJpZM4JIRUn>
> .
>